node_modules/oae-mixpanel/lib/api.js

```diff
@@ -14,24 +14,132 @@
  */
 
 var _ = require('underscore');
+var crypto = require('crypto');
 var Mixpanel = require('mixpanel');
+var MixpanelExporter = require('mixpanel-data-export');
+var moment = require('moment');
+
+var exporter = null;
 
 /**
  * Initialize mixpanel event tracking
  *
  * @param  {Object}     config      An object containing the full system configuration (i.e., `config.js`)
  */
-var init = module.exports.init = function(config) {
+var init = module.exports.init = function(_config) {
+    config = _config;
+
     if (config.mixpanel && config.mixpanel.enabled) {
         // Initialize a client
         var client = Mixpanel.init(config.mixpanel.token);
 
+        // Create a new exporter
+        exporter = new MixpanelExporter({
+            'api_key': config.mixpanel.apikey,
+            'api_secret': config.mixpanel.apisecret
+        });
+
         // Register all the listeners
         _registerListeners(client, config);
     }
 };
 
 /**
+ * Get the unique user count stats for the current month
+ *
+ * @param  {Context}    ctx                 Standard context object containing the current user and the current tenant
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error that occurred, if any
+ * @param  {Object}     callback.stats      Object containing the stats for the tenant(s)
+ */
+var getUniqueUsers = module.exports.getUniqueUsers = function(ctx, callback) {
+    if (ctx.user() && !ctx.user().isAdmin()) {
+        return callback({'code': 401, 'msg': 'Only administrators can request stats'});
+    }
+
+    // Request options object
+    var data = {
+        'from_date': _getStartDate(Date.now()),
+        'to_date': moment.utc(Date.now()).format('YYYY-MM-DD'),
+        'event': 'userLoggedOut',
+        'on': 'properties["tenant"]',
+        'type': 'unique'
+    };
+
+    // Return the unique users for a specific tenant if the endpoint was accessed on the tenant router
+    if (ctx.tenant().alias !== 'admin') {
+        data = _.extend(data, {
```

Comment:
Since `_.extend` will mutate `data`, we don't have to re-assign here with `data =`